### PR TITLE
Refactor the repo into RoslynAnalyzers.sln and FxCopAnalyzers.sln

### PR DESCRIPTION
**Pull request by mavasani**
RoslynAnalyzers.sln has all the existing non-FxCop analyzers projects and a new group of `Microsoft.CodeAnalysis.NetAnalyzers` projects/package. Latter has all the merged CA analyzers from individual packages in FxCop analyzers package and will be used for the new .NET5 analyzers packaged into a single analyzer package. A follow-up change will adjust the default severities and enabled by default values for CA rules shipped as part of this package to be much more conservative then the current FxCopAnalyzers package.

FxCopAnalyzers.sln is the new solution for the existing FxCopAnalyzers package and all its subpackages - Microsoft.CodeQuality.Analyzers, Microsoft.NetCore.Analyzers and Microsoft.NetFramework.Analyzers. There are no changes to the contained CA rules in these packages, and even the default severities and enabled by default values will be preserved. This allows us to continue shipping the existing analyzer packages in the current structure until we are ready to completely migrate from FxCopAnalyzers package to Microsoft.CodeAnalysis.NetAnalyzers package.

**NOTE:** FxCopAnalyzers.sln is just present to generate current packages and VSIXes. RoslynAnalyzers.sln will be continue to be the primary solution for all the development in the repo. You should not need to make any changes/updates to FxCopAnalyzers.sln as part of your development workflow.

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf [hunk 1]
```diff
@@ -2,9 +2,59 @@
 <xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
   <file datatype="xml" source-language="en" target-language="cs" original="../MicrosoftCodeQualityAnalyzersResources.resx">
     <body>
+      <trans-unit id="AbstractTypesShouldNotHaveConstructorsDescription">
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf]: 😕 What happened here?

File: src/Microsoft.CodeAnalysis.NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
+
+Sr. No. | Rule ID | Title | Category | Enabled | CodeFix | Description |
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md]: ❔ Was this supposed to be copied from somewhere?

File: nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.Package.csproj [hunk 3]
```diff
@@ -46,6 +46,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\Microsoft.CodeAnalysis.VersionCheckAnalyzer\Microsoft.CodeAnalysis.VersionCheckAnalyzer.Package.csproj" ReferenceOutputAssembly="false" />
```
Comment (sharwell) [inline nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.Package.csproj]: ❔ Why does this need to be added as part of this pull request?

File: src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf]: I believe moving the resource files somehow caused the xlf files to get regenerated. I am going to investigate if I can avoid this change.

File: src/Microsoft.CodeAnalysis.NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md (hunk 2 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md]: This gets auto-generated for all the analyzer NuGet projects in the solutions.

File: nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.Package.csproj (hunk 3 above)
Comment (mavasani) [reply nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.Package.csproj]: It was just a bug where the FxCop analyzers package already references the VersionCheckAnalyzer package, but there is no P2P reference. I just noticed it while refactoring.

**Comment by mavasani** [comment]
@sharwell Any idea how to avoid the long_path error on CI? Is there some setting I can provide in azure-pipelines.yml to enable long paths? cibuild.cmd succeeds on my machine with VS2019, but seems to fail on CI. 

> 2020-01-03T01:03:35.9478921Z C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(114,5): error : The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. [F:\workspace\_work\1\s\FxCopAnalyzers.sln]

File: src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/xlf/MicrosoftCodeQualityAnalyzersResources.cs.xlf]: Done, reverted.

**Comment by sharwell** [comment]
@tmat or @jaredpar or @JoeRobich are most likely to know this from Roslyn infrastructure.

**Comment by mavasani** [comment]
@sharwell - thanks. For now, I have worked around it by trimming the folder paths, and CI is green now. This PR is ready to review.

**Comment by tmat** [comment]
@mavasani See https://github.com/dotnet/core-eng/issues/8370 for long paths.

**Comment by mavasani** [comment]
Thank you @sharwell!

**Pull request merged**